flask-server/main.py

```diff
@@ -333,31 +333,22 @@ def add_clink():
 
 @app.route('/apis/fetch-clinks', methods=['GET'])
 def fetch_clinks():
-    resp_token = decode_auth_token(request.headers.get('Authorization'))
-
-    if is_valid_instance(resp_token):
-        clink_ids = list(datastore_client.query(kind='user_read_map').add_filter('user_id', '=', int(resp_token)).fetch())
-        all_query = datastore_client.query(kind='clink').add_filter('deleted', '=', False)
-        all_query.order = ['created']
-        all_list = list(all_query.fetch())
-        to_return = []    
-        for clink in all_list:
-            for id in clink_ids:
-                if id['clink_id'] == clink.id:
-                    response_object = {
-                        'title': clink['title'],
-                        'id': clink.id,
-                        'private': clink['private'] 
-                    }
-                    to_return.append(response_object)
-                    break
-        return jsonify(to_return), 200
-    else:
-        response_object = {
-            'status': 'fail',
-            'message': 'Invalid JWT. Failed to fetch clinks.'
-        }
-        return response_object, 401
+    clink_ids = list(datastore_client.query(kind='user_read_map').add_filter('user_id', '=', int(request.headers.get('UserId'))).fetch())
```

Comment:
Why did you get rid of authorization?

Reply:
needed to change to fetching clink with user id instead of token to to be able to view other people's clinks

Reply:
Authorization isn't just to get the user's if.

Reply:
also index.js in users checks the user and redirects to login page if the token is expired, so there's no need to check the token again for fetching clinks

Reply:
In theory you can access the API not directly through websitem

Reply:
If we want to be able to see user pages without being logged in, which we did during the product design stage, then we probably don't want to check for authorization in every call. Not sure if it would be too much work to implement the logged out version though, since everything we've done up to this point has been for logged in users only.